### PR TITLE
Build dir abstract

### PR DESCRIPTION
**Pull request by rgrinberg**
@diml this is a subset of #725. It will allow to customize the build dir in the (near) future without breaking too much code and reduce the rebasing required to keep #725 up to date.

### CODE REVIEW

**Comment by diml** [comment]
If we switch to symbolic paths, we don't need the `unit ->` anymore right?

**Comment by rgrinberg** [comment]
Yeah, i'll edit that commit out.

**Comment by diml** [comment]
The paths constants are still of type `unit -> Path.t`

**Comment by rgrinberg** [comment]
Those are done now as well. Sorry.

File: src/paths.mli [hunk 1]
```diff
@@ -0,0 +1,24 @@
+open Stdune
+
+val digest_db : Path.t
+
+val db : Path.t
+
+val log : Path.t
+
+val to_delete_source_tree : Path.t
```
Comment (diml) [inline src/paths.mli]: `to_delete_in_source_tree`?

**Comment by diml** [comment]
Ok, one last question, do you still think it's better to move all the paths to a single module? I don't have a strong opinion but it seems slightly better to me if they are defined in their respective module.

**Comment by rgrinberg** [comment]
It was done before to create them with the `private` function. That is no longer necessary. I don't have a strong opinion as long as we don't duplicate the string literals. I'll revert it if you prefer the old way.

**Comment by rgrinberg** [comment]
#738 does this in 1 commit.

**Pull request closed without merging**